scripts/hud/customizer.ts

```diff
@@ -40,6 +40,40 @@ namespace HudCustomizer {
 		TOP_LEFT: 8
 	};
 
+	enum DragMode {
+		MOVE = 0,
+		RESIZE_TOP = 1,
+		RESIZE_TOP_RIGHT = 2,
+		RESIZE_RIGHT = 3,
+		RESIZE_BOTTOM_RIGHT = 4,
+		RESIZE_BOTTOM = 5,
+		RESIZE_BOTTOM_LEFT = 6,
+		RESIZE_LEFT = 7,
+		RESIZE_TOP_LEFT = 8
+	}
+
+	const resizeVector: ReadonlyMap<Exclude<DragMode, DragMode.MOVE>, [number, number]> = new Map([
+		[DragMode.RESIZE_TOP, [0, -1]],
+		[DragMode.RESIZE_TOP_RIGHT, [1, -1]],
+		[DragMode.RESIZE_RIGHT, [1, 0]],
+		[DragMode.RESIZE_BOTTOM_RIGHT, [1, 1]],
+		[DragMode.RESIZE_BOTTOM, [0, 1]],
+		[DragMode.RESIZE_BOTTOM_LEFT, [-1, 1]],
+		[DragMode.RESIZE_LEFT, [-1, 0]],
+		[DragMode.RESIZE_TOP_LEFT, [-1, -1]]
+	]);
+
+	function getMinSize(panel: Panel): [number, number] {
+		function parsePx(str: string): number | undefined {
+			if (str && str.slice(-2) === 'px') {
+				return Number.parseFloat(str.slice(0, -2));
+			}
+			return undefined;
+		}
```

Comment:
Don't love nested functions like this, just as easy to read if `parsePx` is separate, and theoretically this is slower since `parsePx` captures `panel` (though in practice in V8 it's probably completely fine).

Reply:
Done